### PR TITLE
Add optional e2e create job for k/cp-gcp.

### PR DESCRIPTION
**Pull request by amwat**
/cc @BenTheElder @cheftako

### CODE REVIEW

**Comment by amwat** [comment]
xref: https://github.com/kubernetes/cloud-provider-gcp/pull/144

File: config/jobs/kubernetes/cloud-provider-gcp/cloud-provider-gcp-presubmits.yaml [hunk 1]
```diff
@@ -25,3 +25,21 @@ presubmits:
         - --
         - //...
         - -//vendor/...
+  - name: cloud-provider-gcp-e2e-create
+    max_concurrency: 5
```
Comment (BenTheElder) [inline config/jobs/kubernetes/cloud-provider-gcp/cloud-provider-gcp-presubmits.yaml:29]: :man_shrugging:

File: config/jobs/kubernetes/cloud-provider-gcp/cloud-provider-gcp-presubmits.yaml [hunk 2]
```diff
@@ -25,3 +25,21 @@ presubmits:
...
+      preset-k8s-ssh: "true"
+    annotations:
+      testgrid-dashboards: provider-gcp-presubmits
+      description: End to end cluster creation test using cluster/kube-up based on kubernetes/cloud-provider-gcp.
+      testgrid-num-columns-recent: '30'
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20200420-e830a3a-master
+        command:
+        - runner.sh
```
Comment (BenTheElder) [inline config/jobs/kubernetes/cloud-provider-gcp/cloud-provider-gcp-presubmits.yaml:43]: is this using anything from runner? the service account?

File: config/jobs/kubernetes/cloud-provider-gcp/cloud-provider-gcp-presubmits.yaml (hunk 1 above)
Comment (amwat) [reply config/jobs/kubernetes/cloud-provider-gcp/cloud-provider-gcp-presubmits.yaml]: removed.

File: config/jobs/kubernetes/cloud-provider-gcp/cloud-provider-gcp-presubmits.yaml (hunk 2 above)
Comment (amwat) [reply config/jobs/kubernetes/cloud-provider-gcp/cloud-provider-gcp-presubmits.yaml:41]: yes

**Pull request merged**